test/contrib/dropbox_test.py

```diff
@@ -0,0 +1,284 @@
+        self.dropbox_api.files_upload(b'hello', DROPBOX_TEST_SIMPLE_FILE)
+        self.dropbox_api.files_upload(b'hello2', DROPBOX_TEST_FILE_IN_DIR)
+        self.dropbox_api.files_upload(b'hello3', DROPBOX_TEST_FILE_TO_MOVE_ORIG)
+        self.dropbox_api.files_upload(b'hello4', DROPBOX_TEST_FILE_TO_COPY_ORIG)
+        self.dropbox_api.files_upload(b'hello3.1', DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.dropbox_api.files_upload(b'hello3.2', DROPBOX_TEST_FILE_TO_DELETE_2)
+
+    def tearDown(self):
+        self.dropbox_api.files_delete_v2(DROPBOX_TEST_PATH)
+        self.dropbox_api._session.close()
+
+    def test_exists(self):
+        self.assertTrue(self.luigiconn.exists('/'))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_PATH))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_DIR))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_SIMPLE_FILE))
+
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_listdir_simple(self):
+        list = self.luigiconn.listdir(DROPBOX_TEST_PATH)
+        self.assertTrue('/' not in list)
+        self.assertTrue(DROPBOX_TEST_PATH in list)
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_listdir_multiple(self):
+        list = self.luigiconn.listdir(DROPBOX_TEST_PATH, limit=2)
+        self.assertTrue('/' not in list)
+        self.assertTrue(DROPBOX_TEST_PATH in list)
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_remove(self):
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+        self.luigiconn.remove(DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+        self.luigiconn.remove(DROPBOX_TEST_DIR_TO_DELETE)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+    def test_remove_nonexisting(self):
+        self.assertFalse(self.luigiconn.remove(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_mkdir_new_dir(self):
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_DIR_TO_CREATE))
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_OUTER_DIR_TO_CREATE))
+        self.luigiconn.mkdir(DROPBOX_TEST_DIR_TO_CREATE)
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_OUTER_DIR_TO_CREATE))
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_DIR_TO_CREATE))
+
+    def test_mkdir_recreate_dir(self):
+        try:
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_DIR)
+        except Exception as ex:
+            self.fail("mkdir with default options raises Exception:" + str(ex))
+
+        try:
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_DIR, raise_if_exists=False)
```

Comment:
if this is `raise_if_exists=False` then we shouldn't hit the exception, right? if  so, why wrap in try/except?

Reply:
and this is the same as the above too (since False is the default value for raise_if_exists)

Reply:
> if this is raise_if_exists=False then we shouldn't hit the exception, right? if so, why wrap in try/except?

I wanted to check that raise_if_exists=False did not raised an exception. If and exception is raised (due to a wrongly coded function), then I would get a Failed Test (with an explanatory message) instead of an error (with a cryptic stacktrace).

> and this is the same as the above too (since False is the default value for raise_if_exists)
I wanted to check the function with and without an value explicit raise_if_exists. This allows me to verify that the function has been indeed implemented with a default value of raise_if_exists=False